### PR TITLE
Add snippet field to AirMapMarker

### PR DESCRIPTION
**Pull request by jvartanian94**
Add ability to set a snippet to an AirMapMarker

### CODE REVIEW

File: library/src/main/java/com/airbnb/android/airmapview/AirMapMarker.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
 package com.airbnb.android.airmapview;
 
+import android.graphics.Bitmap;
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/AirMapMarker.java]: `Bitmap` isn't used, can remove this import

File: sample/src/main/java/com/airbnb/airmapview/sample/MainActivity.java [hunk 2]
```diff
@@ -94,16 +94,17 @@
...
-    addMarker("Performance Bikes", new LatLng(37.773975,-122.40205));
-    addMarker("REI", new LatLng(37.772127, -122.404411));
+    addMarker("Airbnb HQ", "snippet", airbnbLatLng);
+    addMarker("Performance Bikes", "snippet", new LatLng(37.773975, -122.40205));
+    addMarker("REI", "snippet", new LatLng(37.772127, -122.404411));
     map.animateCenterZoom(airbnbLatLng, 10);
   }
 
-  private void addMarker(String title, LatLng latLng) {
+  private void addMarker(String title, String snippet, LatLng latLng) {
```
Comment (petzel) [inline sample/src/main/java/com/airbnb/airmapview/sample/MainActivity.java]: for the sample, can you overload `addMarker` to keep the 2 argument method, and only use the method to include the snippet if it is actually used? we're setting the snippet here but aren't using it for anything - it's the same value for each marker

File: library/src/main/java/com/airbnb/android/airmapview/AirMapMarker.java (hunk 1 above)
Comment (jvartanian94) [inline library/src/main/java/com/airbnb/android/airmapview/AirMapMarker.java]: @petzel Bitmap is used to set the icon of the marker

Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/AirMapMarker.java]: ah it was just moved, got it

**Comment by petzel** [comment]
The snippet is only used in the native map - we'll have to add logic to add this field to the web map as well.

**Comment by jvartanian94** [comment]
@petzel  I was looking at the info window implementation for the web map earlier. The web map doesn't show an info window by default like the native map. I was planning on removing the current info window implementation for the web based map and implementing a default window if the title and/or snippet is set. Wanted to get your thoughts on it first.

**Comment by nwadams** [comment]
@jvartanian94 that sounds like a good approach that we can use for a 'default' info window

File: library/src/main/assets/google_map.html [hunk 3]
```diff
@@ -90,6 +92,26 @@
       icon: 'http://maps.google.com/mapfiles/ms/icons/red-dot.png'
   });
 
+  if (title == "null" && snippet == "null") {
+        <!-- Title and snippet not set, don't create default InfoWindow -->
+  } else if (title != "null" && snippet == "null") {
```
Comment (nwadams) [inline library/src/main/assets/google_map.html]: Could this block of logic be simplified to 
```
var content = document.createElement("div");
content.innerHTML = '';
if (title != "null") {
  content.innerHTML += '<div id="title"><b>' + title + '</b></div>';
}

if (snippet != "null") {
  content.innerHTML += '<div id="snippet">' + snippet + '</div>';
}

infoWindowContent[id] = content;
```

so we wouldn't have three blocks? Forgive me if my lack of understanding of how the innerHTML is built prevents us from combining the blocks.

Comment (jvartanian94) [inline library/src/main/assets/google_map.html]: @nwadams Well to emulate the native map exactly, we don't want to create an InfoWindow if no title and snippet is set. That is why I wrote the logic that way. With your suggestion, an infoWindow will be created every time, unless we change the logic in the showDefaultInfoWindow function.

Comment (nwadams) [inline library/src/main/assets/google_map.html]: @jvartanian94 could we wrap the entire block in a check to ensure that at least one of them in not null.

**Comment by nwadams** [comment]
I'm +1 on this. @petzel any objections to merging?

File: sample/src/main/java/com/airbnb/airmapview/sample/MainActivity.java [hunk 4]
```diff
@@ -94,16 +94,23 @@
...
-    map.addMarker(new AirMapMarker(latLng, 1)
-                      .setTitle(title)
-                      .setIconId(R.mipmap.icon_location_pin));
+  private void addMarker(String title, LatLng latLng, int id) {
+    map.addMarker(new AirMapMarker(latLng, id)
+            .setTitle(title)
+            .setIconId(R.mipmap.icon_location_pin));
+  }
+
+  private void addMarker(String title, String snippet, LatLng latLng, int id){
```
Comment (petzel) [inline sample/src/main/java/com/airbnb/airmapview/sample/MainActivity.java]: can you remove this unused method

**Pull request merged**